packages/kbn-build/src/utils/Package.js

```diff
@@ -0,0 +1,130 @@
+import path from 'path';
+import semver from 'semver';
+import chalk from 'chalk';
+
+import {
+  runScriptInDir,
+  runScriptInPackageStreaming,
+  installInDir
+} from './npm';
+import { readPackageJson } from './packageJson';
+import { CliError } from './errors';
+
+export class Package {
+  static async fromPath(path) {
+    const pkgJson = await readPackageJson(path);
+    return new Package(pkgJson, path);
+  }
+
+  constructor(packageJson, packageDir) {
+    this._json = packageJson;
+    this.path = packageDir;
+
+    this.packageJsonLocation = path.resolve(this.path, 'package.json');
+    this.nodeModulesLocation = path.resolve(this.path, 'node_modules');
+    this.targetLocation = path.resolve(this.path, 'target');
+  }
+
+  get name() {
+    return this._json.name;
+  }
+
+  get version() {
+    return this._json.version;
+  }
+
+  get dependencies() {
+    return this._json.dependencies || {};
+  }
+
+  get devDependencies() {
+    return this._json.devDependencies || {};
+  }
+
+  get peerDependencies() {
+    return this._json.peerDependencies || {};
+  }
+
+  get allDependencies() {
```

Comment:
__note:__ maybe that's just me, but having `peerDependencies` and `allDependencies` that doesn't include them looks a bit confusing :) I don't have strong opinion on that, so up to you.

Reply:
Got rid of `peerDependencies`, ended up never needing them (at least yet).